testsuite/features/step_definitions/salt_steps.rb

```diff
@@ -279,11 +279,14 @@
   # Complicated code because the checkbox is not a <input type=checkbox> but an <i>
   xpath_query = "//a[@id = '#{formula}']/i[@class = 'fa fa-lg fa-square-o']" if action == 'check'
   xpath_query = "//a[@id = '#{formula}']/i[@class = 'fa fa-lg fa-check-square-o']" if action == 'uncheck'
+  puts "Selected formulas [beginning]:\n#{find('#chooseFormulas')['innerHTML']}"
   if all(:xpath, xpath_query, wait: DEFAULT_TIMEOUT).any?
+    puts "Selected formulas [if]:\n#{find('#chooseFormulas')['innerHTML']}"
```

Comment:
I don't get what's the difference of content between the `find('#chooseFormulas')['innerHTML']` before and after the `if`

Reply:
This is exactly what we need to investigate ;-) Of course you're right - there shouldn't be any difference. I'd like to be absolutely sure about content in DOM in those 3 precise moments.

Reply:
Perfect then :)
In addition, you might want to add an assert comparing previous and new content.

Reply:
(Btw. because of `wait` in all method there could be difference even in those infos.)